File: Py Scripts/Playercontrol.py
import UnityEngine

class PlayerControl(UnityEngine.MonoBehaviour):
    speed = 0.0
    up = None
    down = None
    rb = None
    isPlayer = True
    ball = None
    offset = 0.2

    def Start(self):
        self.rb = self.GetComponent(UnityEngine.Rigidbody)

    def Update(self):
        if self.isPlayer:
            self.MoveByPlayer()
        else:
            self.MoveByComputer()

    def MoveByPlayer(self):
        pressedUp = UnityEngine.Input.GetKey(self.up)
        pressedDown = UnityEngine.Input.GetKey(self.down)

        if pressedUp:
            self.rb.velocity = UnityEngine.Vector3.forward * self.speed

        if pressedDown:
            self.rb.velocity = UnityEngine.Vector3.back * self.speed

        if not pressedUp and not pressedDown:
            self.rb.velocity = UnityEngine.Vector3.zero

    def MoveByComputer(self):
        if self.ball.position.z > self.transform.position.z + self.offset:
            self.rb.velocity = UnityEngine.Vector3.forward * self.speed
        elif self.ball.position.z < self.transform.position.z - self.offset:
            self.rb.velocity = UnityEngine.Vector3.back * self.speed
        else:
            self.rb.velocity = UnityEngine.Vector3.zero

File: scripts/Playercontrol.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Playercontrol : MonoBehaviour
{
    public float speed;

    public KeyCode up;

    public KeyCode down;

    private Rigidbody rb;
    
    public bool isPlayer = true;

    public Transform ball;

    public float offset = 0.2f;

    // Start is called before the first frame update
    void Start()
    {
        rb = GetComponent<Rigidbody>();
       
    }

    // Update is called once per frame
    void Update()
    {
        if (this.isPlayer)
        {
            MoveByPlayer();

        }
        else
        {
            MoveByComputer();
        }
    }
    private void MoveByPlayer()
    {

        bool pressedUp = Input.GetKey(this.up);
        bool pressedDown = Input.GetKey(this.down);

        if (pressedUp)
        {
            rb.velocity = Vector3.forward * speed;
        }

        if (pressedDown)
        {
            rb.velocity = Vector3.back * speed;
        }

        if (!pressedUp && !pressedDown)
        {
            rb.velocity = Vector3.zero;
        }
    }
    private void MoveByComputer()
    {
        if(ball.position.z > transform.position.z + offset )
        {
            rb.velocity = Vector3.forward * speed;

        }
        else if(ball.position.z < transform.position.z - offset)
        {
            rb.velocity = Vector3.back * speed;
        }
        else { rb.velocity = Vector3.zero; }
    }
}
